File: c++/PC.cpp
#include<iostream>
#include<string>

using namespace std;
#include "Disk.cpp"
#include "CPU.cpp"
#include "Ram.cpp"

class PC{


    private :
    Disk dObject;
    Ram rObject;
    CPU cObject;
    int total_price;

    //  konstruktor
    public :
    PC(){

    }

    PC(Disk d, Ram r, CPU c){
        dObject = d;
        rObject = r;
        cObject = c;
        total_price = dObject.getDiskPrice() + rObject.getRamPrice()+cObject.getPriceCpu();
    }


    //setter getter
    void setCPU(CPU c){
        cObject = c;
    }

    void setRam(Ram r){
        rObject = r;
    }

    void setDisk(Disk d){
        dObject = d;
    }

    void setTotalPrice(Disk d, Ram r, CPU c){
        total_price = d.getDiskPrice() + r.getRamPrice()+ c.getPriceCpu();
    }

    Disk getDisk(){
        return dObject;
    }

    Ram getRam(){
        return rObject;
    }

    CPU getCPU(){
        return cObject;
    }

    int getTotalPrice(){
        return total_price;
    }

    

    ~PC(){

    }
};

File: PHP/PC.php
<?php
  include "CPU.php";
  include "Disk.php";
  include "Ram.php";

class PC{


    
    private $c;
    private $d;
    private $r;
    private $total_price;
    
   

    public function __construct(CPU $c, Disk $d , Ram $r) {
    $this->cObject = $c = new CPU();
    $this->dObject = $d = new Disk();
    $this->rObject = $r = new RAM();
    $this->total_price =  $c->getCPUprice() + $d->getDiskprice() + $r->getRamprice();

    }

    public function setCPU($c) { //
        $this->cObject = $c;
    }

    public function getCPU() { 
        return $this->cObject;
    }

    public function setDisk($disk) { //
        $this->dObject = $disk;
    }

    public function getDisk() { 
        return $this->dObject;
    }

    public function setRam($ram) { //
        $this->rObject = $ram;
    }

    public function getRam(){ 
        return $this->rObject;

    } 
    public function setTotalPrice($c , $d , $r ){
        $this->total_price =  $c->getCPUprice() + $d->getDiskprice() + $r->getRamprice();
    }

    public function getTotalPrice(){
        return $this->total_price;
    }

}
?>